jsonrpc-specs/src/methods/state.json

```diff
@@ -0,0 +1,490 @@
+          }
+        }
+      },
+      {
+        "name": "requestId",
+        "description": "request-id element of message-data",
+        "required": true,
+        "schema": {
+          "description": "Hex encoded RLP byte array of length <= 8 bytes",
+          "$ref": "#/components/schemas/bytes8"
+        }
+      }
+    ],
+    "result": {
+      "name": "stateSendNodesResult",
+      "description": "Returns the number of batches in which the given ENRs were divided and transmitted",
+      "schema": {
+        "title": "number of batches",
+        "type": "number"
+      }
+    }
+  },
+  {
+    "name": "portal_statePing",
+    "summary": "Send a PING message to the designated node and wait for a PONG response.",
+    "params": [
+      {
+        "name": "enr",
+        "required": true,
+        "schema": {
+          "title": "Ethereum node record",
+          "$ref": "#/components/schemas/Enr"
+        }
+      },
+      {
+        "name": "payload",
+        "description": "Custom payload data",
+        "schema": {
+          "$ref": "#/components/schemas/hexString"
+        }
+      }
+    ],
+    "result": {
+      "name": "statePingResult",
+      "description": "Returns PONG response",
+      "schema": {
+        "title": "PONG message",
+        "type": "object",
+        "required": [
+          "enrSeq",
+          "recipientIP",
+          "recipientPort"
+        ],
+        "properties": {
+          "enrSeq": {
+            "description": "ENR sequence number of sender",
+            "type": "number"
+          },
+          "recipientIP": {
+            "description": "IP address of the intended recipient",
+            "pattern": "((^\\s*((([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5]))\\s*$)|(^\\s*((([0-9A-Fa-f]{1,4}:){7}([0-9A-Fa-f]{1,4}|:))|(([0-9A-Fa-f]{1,4}:){6}(:[0-9A-Fa-f]{1,4}|((25[0-5]|2[0-4]\\d|1\\d\\d|[1-9]?\\d)(\\.(25[0-5]|2[0-4]\\d|1\\d\\d|[1-9]?\\d)){3})|:))|(([0-9A-Fa-f]{1,4}:){5}(((:[0-9A-Fa-f]{1,4}){1,2})|:((25[0-5]|2[0-4]\\d|1\\d\\d|[1-9]?\\d)(\\.(25[0-5]|2[0-4]\\d|1\\d\\d|[1-9]?\\d)){3})|:))|(([0-9A-Fa-f]{1,4}:){4}(((:[0-9A-Fa-f]{1,4}){1,3})|((:[0-9A-Fa-f]{1,4})?:((25[0-5]|2[0-4]\\d|1\\d\\d|[1-9]?\\d)(\\.(25[0-5]|2[0-4]\\d|1\\d\\d|[1-9]?\\d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){3}(((:[0-9A-Fa-f]{1,4}){1,4})|((:[0-9A-Fa-f]{1,4}){0,2}:((25[0-5]|2[0-4]\\d|1\\d\\d|[1-9]?\\d)(\\.(25[0-5]|2[0-4]\\d|1\\d\\d|[1-9]?\\d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){2}(((:[0-9A-Fa-f]{1,4}){1,5})|((:[0-9A-Fa-f]{1,4}){0,3}:((25[0-5]|2[0-4]\\d|1\\d\\d|[1-9]?\\d)(\\.(25[0-5]|2[0-4]\\d|1\\d\\d|[1-9]?\\d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){1}(((:[0-9A-Fa-f]{1,4}){1,6})|((:[0-9A-Fa-f]{1,4}){0,4}:((25[0-5]|2[0-4]\\d|1\\d\\d|[1-9]?\\d)(\\.(25[0-5]|2[0-4]\\d|1\\d\\d|[1-9]?\\d)){3}))|:))|(:(((:[0-9A-Fa-f]{1,4}){1,7})|((:[0-9A-Fa-f]{1,4}){0,5}:((25[0-5]|2[0-4]\\d|1\\d\\d|[1-9]?\\d)(\\.(25[0-5]|2[0-4]\\d|1\\d\\d|[1-9]?\\d)){3}))|:)))(%.+)?\\s*$))"
```

Comment:
Is there a way for us to de-duplicate the definition of this regular expression by using references?